### PR TITLE
Override keyword args for automatically generated serializer fields

### LINKED ISSUE #98: Support overriding serializer arguments for automatically generated serializers
For our schema generation support, we currently support automatic generation of serializer fields via model introspection:

```py
spec = [
    "title",
]

# -->

class SomeSerializer(Serializer):
    title = CharField()
```

We also support overriding the auto-generated serializer field, using `out`:

```py
spec = [
    "title" >> out(serializers.CharField(label="The title")),
]

# -->

class SomeSerializer(Serializer):
    title = CharField(label="The title")
```

However, in a large spec, if you wish to provide `label` and/or `help_text` (which are extremely useful during schema generation, as [drf-spectacular uses them for the "title" and "description" OpenAPI attributes](https://github.com/tfranzel/drf-spectacular/blob/a79bc5d68af5461157bd229429cbb6c1a926ba8b/drf_spectacular/openapi.py#L1007-L1010)) to multiple fields in a large spec, using `out` is tedious, because you are essentially throwing away all of the clever automatic generation of serializer fields, and doing everything manually.

It would be nice if it were possible to provide extra keyword arguments to the default (automatically created) serializer fields.

As we already use the `>>` syntax for `out`, we could use the same idea. A couple of possibilities:

```py
spec = [
    "title" >> kwargs(label="The title", help_text="Some help"),
]
```

Or perhaps just:

```py
spec = [
    "title" >> label("The title") >> help_text("Some help"),
]
```

`kwargs` feels a little awkward because it implies that _other_ kwargs than just `label` and `help_text` can be supplied - which is technically true, but none of the others are used by drf-spectacular in a read-only context, and remember the serializer we're generating here is only used for schema generation, not for actually serializing the objects (but then nothing else here is specifically tied to drf-spectacular, so maybe we should be thinking more generally?). On the other hand, providing `label` and `help_text` separately feels a bit awkward. Any other ideas? Any other arguments we might want to provide?

### PR DESCRIPTION
**Pull request by j4mie**
Fixes #98

This adds optional keyword arguments to the `out` object (when used as either a decorator or using the `>>` approach) which are passed to the constructor of the serializer field. This provides a way to add useful attributes to the field (like `label` and `help_text`) without throwing away the automagic model introspection behaviour.

```py
class SomeView(SpecMixin, RetrieveAPIView):
    queryset = SomeModel.objects.all()
    spec = [
        ...,
        "title" >> out(label="The title of the object")
        ...,
    ]
```

Thanks @carltongibson for the API suggestion.

### CODE REVIEW

**Comment by j4mie** [comment]
Just calling out that this does change an internal API: previously, the serializer field was added to an attribute named `out`, now we have a generic `_readers_annotation` dictionary. This is undocumented behaviour so does not require a major version bump.

File: django_readers/rest_framework.py [hunk 1]
```diff
@@ -231,22 +244,28 @@ def serializer_class_for_view(view):
...
-        if not all(
-            isinstance(item, serializers.Field) for item in field_or_dict.values()
-        ):
-            raise TypeError("Each value must be an instance of Field")
-    elif not isinstance(field_or_dict, serializers.Field):
-        raise TypeError("Must be an instance of Field")
+def out(*args, **kwargs):
+    if args:
+        if len(args) != 1:
+            raise TypeError("Provide a single field or dictionary of fields")
```
Comment (carltongibson) [inline django_readers/rest_framework.py:258]: Here, it may be worth checking that `kwargs` are empty too. 🤔

Comment (j4mie) [reply django_readers/rest_framework.py:258]: I've decided to explicitly support passing _both_ a field, and also kwargs. See the tests for some examples. I know it's a redundant thing to do (you could just as easily pass the kwargs into the field constructor directly), but it felt more consistent and less surprising to have it "just work"...

Comment (carltongibson) [reply django_readers/rest_framework.py:258]: Great. No problem 👍

**Pull request merged**